File: app/src/main/java/com/arkapp/partyplanner/ui/home/HomeFragment.java
// HomeFragment.java
package com.arkapp.partyplanner.ui.home;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.os.AsyncTask;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.activity.OnBackPressedCallback;
import androidx.fragment.app.Fragment;

import com.arkapp.partyplanner.R;
import com.arkapp.partyplanner.data.models.PartyDetails;
import com.arkapp.partyplanner.data.repository.PrefRepository;
import com.arkapp.partyplanner.data.room.AppDatabase.Companion;
import com.arkapp.partyplanner.data.room.UnfinishedDetailsDao;
import com.arkapp.partyplanner.databinding.FragmentHomeBinding;
import com.arkapp.partyplanner.utils.AddUnfinishedAsyncTask;
import com.arkapp.partyplanner.utils.AppDataKt;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;

import kotlin.jvm.internal.Intrinsics;

import static androidx.navigation.fragment.NavHostFragment.findNavController;
import static com.arkapp.partyplanner.utils.AppDataKt.OPTION_UNFINISHED;
import static com.arkapp.partyplanner.utils.AppDataKt.getPartyTypeFromStringArray;
import static com.arkapp.partyplanner.utils.AppDataKt.getPartyTypes;
import static com.arkapp.partyplanner.utils.ViewUtilsKt.getDrawableRes;
import static com.arkapp.partyplanner.utils.ViewUtilsKt.initGridAdapter;
import static com.arkapp.partyplanner.utils.ViewUtilsKt.toastShort;


public class HomeFragment extends Fragment {
    private FragmentHomeBinding binding;
    private PartyTypeAdapter partyTypeAdapter;
    private PrefRepository prefRepository;

    @Nullable
    public View onCreateView(@NotNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentHomeBinding.inflate(inflater);
        prefRepository = new PrefRepository(requireContext());

        return binding.getRoot();
    }

    public void onViewCreated(@NotNull View view, @Nullable Bundle savedInstanceState) {
        Intrinsics.checkParameterIsNotNull(view, "view");
        super.onViewCreated(view, savedInstanceState);

        resetPartyData();
        initPartyTypeUI();
        initBudgetBtnListener();
        initDestinationBtnListener();
        initCalendar();

        //Used to store the guest count
        binding.guestEt.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {
                if (!s.toString().isEmpty()) {
                    binding.guest.setError(null);
                    PartyDetails details = prefRepository.getCurrentPartyDetails();
                    details.setPartyGuest(Integer.parseInt(s.toString()));
                    prefRepository.setCurrentPartyDetails(details);
                }
            }
        });

        //Validating the screen input on button click
        binding.proceedBtn.setOnClickListener(v -> {
            if (binding.guestEt.getText().toString().isEmpty() ||
                    Integer.parseInt(binding.guestEt.getText().toString()) <= 0) {
                binding.guest.setError("Please enter guest count!");
                return;
            }

            if (prefRepository.getCurrentPartyDetails().getPartyType().isEmpty()) {
                toastShort(requireContext(), "Please select party type");
                return;
            }

            binding.guest.setError(null);

            new AddUnfinishedAsyncTask(requireActivity(), prefRepository).execute();
            findNavController(this).navigate(R.id.action_homeFragment_to_caterersListFragment);
        });

        if (AppDataKt.getCURRENT_SELECTED_OPTION() == OPTION_UNFINISHED)
            setUnfinishedPartyData();
        else
            setLastEnteredData();

        //Handling back click
        OnBackPressedCallback callback = new OnBackPressedCallback(true) {
            @Override
            public void handleOnBackPressed() {
                // Handle the back button event
                prefRepository.setCurrentPartyDetails(
                        new PartyDetails(
                                null,
                                null,
                                null,
                                null,
                                null,
                                new ArrayList<String>(),
                                null,
                                null,
                                null,
                                null,
                                null,
                                new ArrayList<String>())
                );
                this.remove();
                requireActivity().onBackPressed();
            }
        };
        requireActivity().getOnBackPressedDispatcher().addCallback(getViewLifecycleOwner(), callback);
    }

    //Setting the Unfinished party data in SQL
    private void setUnfinishedPartyData() {
        GetUnfinishedSummaryListener taskListener = (unfinishedDetail -> {
            Date date = unfinishedDetail.getPartyDate();
            if (date != null) {
                Calendar selectedDate = Calendar.getInstance();
                selectedDate.setTime(date);
                binding.calendarView.setDate(selectedDate.getTimeInMillis());
            }

            String budget = unfinishedDetail.getPartyBudget();
            if (budget != null) {
                if (getString(R.string.low).equals(budget)) {
                    binding.lowBudget.performClick();
                } else if (getString(R.string.medium).equals(budget)) {
                    binding.mediumBudget.performClick();
                } else if (getString(R.string.high).equals(budget)) {
                    binding.highBudget.performClick();
                } else if (getString(R.string.very_high).equals(budget)) {
                    binding.veryHighBudget.performClick();
                } else {
                    binding.lowBudget.performClick();
                }
            }

            String destination = unfinishedDetail.getPartyDestination();
            if (destination != null) {
                if (destination.equals(getString(R.string.home))) {
                    binding.homeParty.performClick();
                } else
                    binding.venueParty.performClick();
            }

            Integer guest = unfinishedDetail.getPartyGuest();
            if (guest != null) {
                binding.guestEt.setText(guest.toString());
            }

            ArrayList<String> type = unfinishedDetail.getPartyType();
            partyTypeAdapter.selectedPartyType = getPartyTypeFromStringArray(type);
            partyTypeAdapter.notifyDataSetChanged();
            PartyDetails details = prefRepository.getCurrentPartyDetails();
            details.setPartyType(type);
            prefRepository.setCurrentPartyDetails(details);

        });
        new GetUnfinishedSummaryAsyncTask(requireActivity(), prefRepository, taskListener).execute();
    }

    //Initializing the calendar UI
    private void initCalendar() {
        binding.calendarView.setMinDate(Calendar.getInstance().getTimeInMillis());

        binding.calendarView.setOnDateChangeListener((view, year, month, dayOfMonth) -> {

            Calendar selectedPartyDate = Calendar.getInstance();
            selectedPartyDate.set(Calendar.YEAR, year);
            selectedPartyDate.set(Calendar.MONTH, month);
            selectedPartyDate.set(Calendar.DAY_OF_MONTH, dayOfMonth);

            PartyDetails details = prefRepository.getCurrentPartyDetails();
            details.setPartyDate(selectedPartyDate.getTime());
            prefRepository.setCurrentPartyDetails(details);
        });
    }

    private void initBudgetBtnListener() {
        binding.lowBudget.setOnClickListener(v -> {
            binding.budget.setText(getString(R.string.low));

            binding.lowBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_selected_start));
            binding.mediumBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected));
            binding.highBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected));
            binding.veryHighBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected_end));

            PartyDetails details = prefRepository.getCurrentPartyDetails();
            details.setPartyBudget(getString(R.string.low));
            prefRepository.setCurrentPartyDetails(details);
        });

        binding.mediumBudget.setOnClickListener(v -> {
            binding.budget.setText(getString(R.string.medium));

            binding.lowBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected_start));
            binding.mediumBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_selected));
            binding.highBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected));
            binding.veryHighBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected_end));

            PartyDetails details = prefRepository.getCurrentPartyDetails();
            details.setPartyBudget(getString(R.string.medium));
            prefRepository.setCurrentPartyDetails(details);
        });

        binding.highBudget.setOnClickListener(v -> {
            binding.budget.setText(getString(R.string.high));

            binding.lowBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected_start));
            binding.mediumBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected));
            binding.highBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_selected));
            binding.veryHighBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected_end));

            PartyDetails details = prefRepository.getCurrentPartyDetails();
            details.setPartyBudget(getString(R.string.high));
            prefRepository.setCurrentPartyDetails(details);
        });

        binding.veryHighBudget.setOnClickListener(v -> {
            binding.budget.setText(getString(R.string.very_high));

            binding.lowBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected_start));
            binding.mediumBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected));
            binding.highBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected));
            binding.veryHighBudget.setBackground(getDrawableRes(requireContext(), R.drawable.bg_selected_end));

            PartyDetails details = prefRepository.getCurrentPartyDetails();
            details.setPartyBudget(getString(R.string.very_high));
            prefRepository.setCurrentPartyDetails(details);
        });
    }

    //Initializing the Destination selection UI
    private void initDestinationBtnListener() {
        binding.homeParty.setOnClickListener(v -> {
            binding.homeParty.setBackground(getDrawableRes(requireContext(), R.drawable.bg_selected_start));
            binding.venueParty.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected_end));

            PartyDetails details = prefRepository.getCurrentPartyDetails();
            details.setPartyDestination(getString(R.string.home));
            prefRepository.setCurrentPartyDetails(details);
        });

        binding.venueParty.setOnClickListener(v -> {
            binding.homeParty.setBackground(getDrawableRes(requireContext(), R.drawable.bg_unselected_start));
            binding.venueParty.setBackground(getDrawableRes(requireContext(), R.drawable.bg_selected_end));

            PartyDetails details = prefRepository.getCurrentPartyDetails();
            details.setPartyDestination(getString(R.string.other_venue));
            prefRepository.setCurrentPartyDetails(details);
        });
    }

    //Set the last entered data
    private void setLastEnteredData() {
        PartyDetails detail = prefRepository.getCurrentPartyDetails();

        Date date = detail.getPartyDate();

        if (date != null) {
            Calendar selectedDate = Calendar.getInstance();
            selectedDate.setTime(date);
            binding.calendarView.setDate(selectedDate.getTimeInMillis());

        }

        String budget = detail.getPartyBudget();
        if (budget != null) {
            if (getString(R.string.low).equals(budget)) {
                binding.lowBudget.performClick();
            } else if (getString(R.string.medium).equals(budget)) {
                binding.mediumBudget.performClick();
            } else if (getString(R.string.high).equals(budget)) {
                binding.highBudget.performClick();
            } else if (getString(R.string.very_high).equals(budget)) {
                binding.veryHighBudget.performClick();
            } else {
                binding.lowBudget.performClick();
            }
        }

        String destination = detail.getPartyDestination();
        if (destination != null) {
            if (destination.equals(getString(R.string.home))) {
                binding.homeParty.performClick();
            } else
                binding.venueParty.performClick();
        }

        Integer guest = detail.getPartyGuest();
        if (guest != null) {
            binding.guestEt.setText(guest.toString());
        }

        ArrayList<String> type = detail.getPartyType();
        if (type != null) {
            partyTypeAdapter.selectedPartyType = getPartyTypeFromStringArray(type);
            partyTypeAdapter.notifyDataSetChanged();
        }
    }

    private void initPartyTypeUI() {
        partyTypeAdapter = new PartyTypeAdapter(getPartyTypes(), prefRepository);
        initGridAdapter(binding.partyTypeRv, partyTypeAdapter, true, 3);
    }

    private void resetPartyData() {
        PartyDetails details = new PartyDetails(
                null,
                Calendar.getInstance().getTime(),
                getString(R.string.low),
                getString(R.string.home),
                null,
                new ArrayList<String>(),
                null,
                null,
                null,
                null,
                null,
                new ArrayList<String>()
        );
        prefRepository.setCurrentPartyDetails(details);
    }


    private class GetUnfinishedSummaryAsyncTask extends AsyncTask<Void, Void, PartyDetails> {
        private final Activity context;
        private final PrefRepository prefRepository;
        private final GetUnfinishedSummaryListener taskListener;

        public GetUnfinishedSummaryAsyncTask(@NotNull Activity context, @NotNull PrefRepository prefRepository, @NotNull GetUnfinishedSummaryListener taskListener) {
            super();
            this.context = context;
            this.prefRepository = prefRepository;
            this.taskListener = taskListener;
        }

        @Nullable
        @Override
        protected PartyDetails doInBackground(@NotNull Void... params) {
            UnfinishedDetailsDao unfinishedDao = (new Companion()).getDatabase(context).unfinishedDao();
            return AppDataKt.convertPartyFromUnfinished(unfinishedDao.getUserUnfinished(prefRepository.getCurrentUser().getUid()).get(0));
        }

        @SuppressLint({"SetTextI18n"})
        @Override
        protected void onPostExecute(@Nullable PartyDetails unfinishedSummary) {
            this.taskListener.onTaskEnded(unfinishedSummary);
        }
    }
}

File: app/src/main/java/com/arkapp/partyplanner/ui/home/HomeFragment.kt
package com.arkapp.partyplanner.ui.home

import android.annotation.SuppressLint
import android.app.Activity
import android.os.AsyncTask
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.activity.addCallback
import androidx.core.widget.doAfterTextChanged
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.arkapp.partyplanner.R
import com.arkapp.partyplanner.data.models.PartyDetails
import com.arkapp.partyplanner.data.repository.PrefRepository
import com.arkapp.partyplanner.data.room.AppDatabase
import com.arkapp.partyplanner.databinding.FragmentHomeBinding
import com.arkapp.partyplanner.utils.*
import java.util.*
import kotlin.collections.ArrayList


class HomeFragment : Fragment() {

    private lateinit var binding: FragmentHomeBinding

    private lateinit var partyTypeAdapter: PartyTypeAdapter
    private val prefRepository by lazy { PrefRepository(requireContext()) }

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        //Creating the binding variable which will contain all the view of the layout.
        binding = FragmentHomeBinding.inflate(inflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        resetPartyData()
        initPartyTypeUI()
        initBudgetBtnListener()
        initDestinationBtnListener()
        initCalendar()

        //Used to store the guest count
        binding.guestEt.doAfterTextChanged { text ->
            if (!text.isNullOrEmpty()) {
                binding.guest.error = null
                val details = prefRepository.currentPartyDetails
                details.partyGuest = text.toString().toInt()
                prefRepository.currentPartyDetails = details
            }
        }

        //Validating the screen input on button click
        binding.proceedBtn.setOnClickListener {
            if (binding.guestEt.text.toString().isEmpty() ||
                binding.guestEt.text.toString().toInt() <= 0) {
                binding.guest.error = "Please enter guest count!"
                return@setOnClickListener
            }

            if (prefRepository.currentPartyDetails.partyType.isEmpty()) {
                requireContext().toastShort("Please select party type")
                return@setOnClickListener
            }

            binding.guest.error = null

            AddUnfinishedAsyncTask(requireActivity(), prefRepository).execute()
            findNavController().navigate(R.id.action_homeFragment_to_caterersListFragment)
        }

        if (CURRENT_SELECTED_OPTION == OPTION_UNFINISHED)
            setUnfinishedPartyData()
        else
            setLastEnteredData()

        //Handling back click
        requireActivity()
            .onBackPressedDispatcher
            .addCallback(viewLifecycleOwner) {
                prefRepository.currentPartyDetails = PartyDetails(
                    null,
                    null,
                    null,
                    null,
                    null,
                    ArrayList<String>(),
                    null,
                    null,
                    null,
                    null,
                    null,
                    ArrayList<String>())
                this.remove()
                requireActivity().onBackPressed()
                true
            }
    }

    private class GetUnfinishedSummaryAsyncTask(private val context: Activity,
                                                private val prefRepository: PrefRepository,
                                                val taskListener: GetUnfinishedSummaryListener) : AsyncTask<Void, Void, PartyDetails?>() {

        override fun doInBackground(vararg params: Void?): PartyDetails? {
            val unfinishedDao = AppDatabase.Companion().getDatabase(context).unfinishedDao()
            val unfinishedData = unfinishedDao.getUserUnfinished(prefRepository.currentUser?.uid!!)
            return convertPartyFromUnfinished(unfinishedData[0])
        }

        @SuppressLint("SetTextI18n")
        override fun onPostExecute(unfinisedSummary: PartyDetails?) {
            taskListener.onTaskEnded(unfinisedSummary)
        }
    }


    //Setting the Unfinished party data in SQL
    private fun setUnfinishedPartyData() {
        val taskListener = GetUnfinishedSummaryListener { unfinishedDetail ->
            unfinishedDetail.partyDate.also {
                if (it != null) {
                    val selectedDate = Calendar.getInstance()
                    selectedDate.time = it
                    binding.calendarView.date = selectedDate.timeInMillis
                }
            }

            unfinishedDetail.partyBudget.also {
                if (it != null) {
                    when (it) {
                        getString(R.string.low) -> binding.lowBudget.performClick()
                        getString(R.string.medium) -> binding.mediumBudget.performClick()
                        getString(R.string.high) -> binding.highBudget.performClick()
                        getString(R.string.very_high) -> binding.veryHighBudget.performClick()
                        else -> binding.lowBudget.performClick()
                    }
                }
            }

            unfinishedDetail.partyDestination.also {
                if (it != null) {
                    if (it == getString(R.string.home)) {
                        binding.homeParty.performClick()
                    } else
                        binding.venueParty.performClick()
                }
            }

            unfinishedDetail.partyGuest.also {
                if (it != null) {
                    binding.guestEt.setText(it.toString())
                }
            }

            unfinishedDetail.partyType.also {
                partyTypeAdapter.selectedPartyType = getPartyTypeFromStringArray(it)
                partyTypeAdapter.notifyDataSetChanged()
                val details = prefRepository.currentPartyDetails
                details.partyType = it
                prefRepository.currentPartyDetails = details
            }
        }
        GetUnfinishedSummaryAsyncTask(requireActivity(), prefRepository, taskListener).execute()
    }

    //Initializing the calendar UI
    private fun initCalendar() {
        binding.calendarView.minDate = Calendar.getInstance().timeInMillis

        binding.calendarView.setOnDateChangeListener { _, year, month, dayOfMonth ->

            val selectedPartyDate = Calendar.getInstance()
            selectedPartyDate.set(Calendar.YEAR, year)
            selectedPartyDate.set(Calendar.MONTH, month)
            selectedPartyDate.set(Calendar.DAY_OF_MONTH, dayOfMonth)

            val details = prefRepository.currentPartyDetails
            details.partyDate = selectedPartyDate.time
            prefRepository.currentPartyDetails = details
        }
    }

    //Initializing the Budget UI
    private fun initBudgetBtnListener() {
        binding.lowBudget.setOnClickListener {
            binding.budget.text = getString(R.string.low)

            binding.lowBudget.background = requireContext().getDrawableRes(R.drawable.bg_selected_start)
            binding.mediumBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected)
            binding.highBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected)
            binding.veryHighBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected_end)

            val details = prefRepository.currentPartyDetails
            details.partyBudget = getString(R.string.low)
            prefRepository.currentPartyDetails = details
        }

        binding.mediumBudget.setOnClickListener {
            binding.budget.text = getString(R.string.medium)

            binding.lowBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected_start)
            binding.mediumBudget.background = requireContext().getDrawableRes(R.drawable.bg_selected)
            binding.highBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected)
            binding.veryHighBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected_end)

            val details = prefRepository.currentPartyDetails
            details.partyBudget = getString(R.string.medium)
            prefRepository.currentPartyDetails = details
        }

        binding.highBudget.setOnClickListener {
            binding.budget.text = getString(R.string.high)

            binding.lowBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected_start)
            binding.mediumBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected)
            binding.highBudget.background = requireContext().getDrawableRes(R.drawable.bg_selected)
            binding.veryHighBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected_end)

            val details = prefRepository.currentPartyDetails
            details.partyBudget = getString(R.string.high)
            prefRepository.currentPartyDetails = details
        }

        binding.veryHighBudget.setOnClickListener {
            binding.budget.text = getString(R.string.very_high)

            binding.lowBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected_start)
            binding.mediumBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected)
            binding.highBudget.background = requireContext().getDrawableRes(R.drawable.bg_unselected)
            binding.veryHighBudget.background = requireContext().getDrawableRes(R.drawable.bg_selected_end)

            val details = prefRepository.currentPartyDetails
            details.partyBudget = getString(R.string.very_high)
            prefRepository.currentPartyDetails = details
        }
    }

    //Initializing the Destination selection UI
    private fun initDestinationBtnListener() {
        binding.homeParty.setOnClickListener {
            binding.homeParty.background = requireContext().getDrawableRes(R.drawable.bg_selected_start)
            binding.venueParty.background = requireContext().getDrawableRes(R.drawable.bg_unselected_end)

            val details = prefRepository.currentPartyDetails
            details.partyDestination = getString(R.string.home)
            prefRepository.currentPartyDetails = details
        }

        binding.venueParty.setOnClickListener {
            binding.homeParty.background = requireContext().getDrawable(R.drawable.bg_unselected_start)
            binding.venueParty.background = requireContext().getDrawable(R.drawable.bg_selected_end)

            val details = prefRepository.currentPartyDetails
            details.partyDestination = getString(R.string.other_venue)
            prefRepository.currentPartyDetails = details
        }
    }

    //Set the last entered data
    private fun setLastEnteredData() {
        val detail = prefRepository.currentPartyDetails

        detail.partyDate.also {
            if (it != null) {
                val selectedDate = Calendar.getInstance()
                selectedDate.time = it
                binding.calendarView.date = selectedDate.timeInMillis
            }
        }

        detail.partyBudget.also {
            if (it != null) {
                when (it) {
                    getString(R.string.low) -> binding.lowBudget.performClick()
                    getString(R.string.medium) -> binding.mediumBudget.performClick()
                    getString(R.string.high) -> binding.highBudget.performClick()
                    getString(R.string.very_high) -> binding.veryHighBudget.performClick()
                    else -> binding.lowBudget.performClick()
                }
            }
        }

        detail.partyDestination.also {
            if (it != null) {
                if (it == getString(R.string.home)) {
                    binding.homeParty.performClick()
                } else
                    binding.venueParty.performClick()
            }
        }

        detail.partyGuest.also {
            if (it != null) {
                binding.guestEt.setText(it.toString())
            }
        }

        detail.partyType.also {
            partyTypeAdapter.selectedPartyType = getPartyTypeFromStringArray(it)
            partyTypeAdapter.notifyDataSetChanged()
        }
    }

    private fun initPartyTypeUI() {
        partyTypeAdapter = PartyTypeAdapter(getPartyTypes(), prefRepository)
        binding.partyTypeRv.initGridAdapter(partyTypeAdapter, true, 3)
    }

    private fun resetPartyData() {
        val details = PartyDetails(
            null,
            Calendar.getInstance().time,
            getString(R.string.low),
            getString(R.string.home),
            null,
            ArrayList<String>(),
            null,
            null,
            null,
            null,
            null,
            ArrayList<String>()
        )
        prefRepository.currentPartyDetails = details
    }
}
